xmppserver/src/main/java/org/jivesoftware/openfire/pep/IQPEPHandler.java

```diff
@@ -428,6 +385,31 @@ else if (packet.getType() == IQ.Type.get || packet.getType() == IQ.Type.set) {
         return null;
     }
 
+    /**
+     * Populates the PEPService instance with subscriptions. The subscriptions that
+     * are added to the PEPService are based on the roster of the owner of the PEPService:
+     * every entity that's subscribed to the presence of the owner, is added as a
+     * subscriber of the PEPService.
+     *
+     * This method adds, but does not remove of update existing subscriptions.
+     *
+     * @param pepService The PEPService to be populated with subscriptions.
+     */
+    public void addSubscriptionForRosterItems( final PEPService pepService )
```

Comment:
(have yet to read where this method is called so take this comment with that in mind)
Given that this method can silently fail, surprised that there isn't some sort of return code (even if only `true`/`false`) to indicate success.

Reply:
That's consistently done throughout Openfire - especially the older bits. 

The alternative here would probably be to prevent the PEP service from being created, which in itself would introduce other problems (every user of Openfire implicitly has a PEP service, even if the database structures do not yet exist).